storage/client/src/main/java/com/azure/storage/blob/AppendBlobClientBuilder.java

```diff
@@ -106,47 +128,88 @@ public AppendBlobAsyncClient buildAsyncClient() {
     }
 
     /**
-     * Sets the service endpoint, additionally parses it for information (SAS token, container name)
+     * Sets the service endpoint, additionally parses it for information (SAS token, container name, blob name)
      * @param endpoint URL of the service
      * @return the updated AppendBlobClientBuilder object
      */
     public AppendBlobClientBuilder endpoint(String endpoint) {
         Objects.requireNonNull(endpoint);
+        URL url;
         try {
-            this.endpoint = new URL(endpoint);
-        } catch (MalformedURLException ex) {
-            throw new IllegalArgumentException("The Azure Storage Queue endpoint url is malformed.");
+            url = new URL(endpoint);
+            BlobURLParts parts = URLParser.parse(url);
+            this.endpoint = new URL(parts.scheme() + "://" + parts.host());
+            this.containerName = parts.containerName();
+            this.blobName = parts.blobName();
+        } catch (MalformedURLException | UnknownHostException ex) {
+            throw new IllegalArgumentException("The Azure Storage Blob endpoint url is malformed.");
+        }
+
+        SASTokenCredential credential = SASTokenCredential.fromQuery(url.getQuery());
+        if (credential != null) {
+            this.sasTokenCredential = credential;
         }
 
         return this;
     }
 
     /**
-     * Sets the credentials used to authorize requests sent to the service
-     * @param credentials authorization credentials
+     * Sets the name of the container this client is connecting to.
+     * @param containerName the name of the container
+     * @return the updated AppendBlobClientBuilder object
+     */
+    public AppendBlobClientBuilder containerName(String containerName) {
+        this.containerName = containerName;
+        return this;
+    }
+
+    /**
+     * Sets the name of the blob this client is connecting to.
+     * @param blobName the name of the blob
+     * @return the updated AppendBlobClientBuilder object
+     */
+    public AppendBlobClientBuilder blobName(String blobName) {
+        this.blobName = blobName;
+        return this;
+    }
+
+    /**
+     * Sets the credential used to authorize requests sent to the service
+     * @param credential authorization credential
      * @return the updated AppendBlobClientBuilder object
      */
-    public AppendBlobClientBuilder credentials(SharedKeyCredentials credentials) {
-        this.credentials = credentials;
+    public AppendBlobClientBuilder credential(SharedKeyCredential credential) {
```

Comment:
Is there no base type or interface for credentials? We need all these overloads?

Reply:
Unfortunately they all need different credentials policy to work.